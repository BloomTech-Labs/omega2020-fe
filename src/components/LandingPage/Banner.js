import React from 'react';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import Box from '@material-ui/core/Box';
import Skeleton from '@material-ui/lab/Skeleton';
import CardHeader from '@material-ui/core/CardHeader';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

import BlueButton from '../assets/BlueButton';
import games from '../../images/games.png';

const Banner = (props) => {
  const classes = useStyles();
  const { loading = false } = props;

  let viewPort = false;
  const md = useMediaQuery('(min-width: 1050px)');
  if (viewPort === md) {
    viewPort = true;
  } else {
    viewPort = false;
  }

  return (
    <Box className={classes.root}>
      <Box className={classes.mainContainer}>
        <CardHeader
          avatar={
            loading ? (
              <Skeleton
                className='bannerIMG'
                variant='rect'
                width={600}
                height={600}
              />
            ) : (
              <Box className={classes.headerBox}>
                {(() => {
                  switch (viewPort) {
                    case true:
                      return (
                        <Box style={{ textAlign: 'center' }}>
                          <Typography
                            className={classes.header}
                            style={{ fontSize: '1.3rem' }}
                            id='improve1'
                          >
                            Improve Your Skills
                            <br />
                            By Understanding Game Strategies
                          </Typography>
                          <br />
                          <Typography
                            className={classes.subheader}
                            id='improve2'
                          >
                            Learn to recognize key patterns and methods
                            <br />
                            to improve your speed and skill level
                          </Typography>
                          <br />
                          <br />
                          <br />
                          <br />
                        </Box>
                      );
                    default:
                      return (
                        <Box style={{ textAlign: 'center' }}>
                          <Typography
                            variant='h4'
                            className={classes.header}
                            id='bang'
                          >
                            Improve Your Skills By Understanding Game Strategies
                          </Typography>
                          <br />
                          <Typography
                            variant='h6'
                            className={classes.subheader}
                          >
                            Learn to recognize key patterns and methods to
                            improve your speed and skill level
                          </Typography>
                          <br />
                          <br />
                          <br />
                          <br />
                        </Box>
                      );
                  }
                })()}
              </Box>
            )
          }
        />
        <CardHeader
          action={
            loading ? (
              <Skeleton className={classes.text} width={60} height={55} />
            ) : (
              <Box>
                <BlueButton title={'Get Started Today'} href={'/tutorial'} />
              </Box>
            )
          }
        />

        <CardHeader
          action={
            loading ? (
              <Skeleton className={classes.text} width={60} height={55} />
            ) : (
              <Box className={classes.gameBoard}>
                {(() => {
                  switch (viewPort) {
                    case true:
                      return (
                        <Box className={classes.img}>
                          {/* <img src={games} alt='games' width='300' /> */}
                        </Box>
                      );
                    default:
                      return (
                        <Box className={classes.img}>
                          {/* <img src={games} alt='games' /> */}
                        </Box>
                      );
                  }
                })()}
              </Box>
            )
          }
        />
      </Box>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 100,
  },
  mainContainer: {
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'left',
  },
  subheader: {
    // opacity: 0.3,
    color: '#A2A2A2',
  },
  img: {
    width: '90vw',
    height: '50vh',
    backgroundImage: `url(${games})`,
    backgroundRepeat: 'no-repeat',
    // backgroundSize: 'contain, cover',
    backgroundSize: 'cover',
    backgroundPosition: 'center',
  },
  gameBoard: {
    marginTop: 200,
  },
  space: {
    '& > *': {
      margin: theme.spacing(1),
    },
  },
}));

export default Banner;
