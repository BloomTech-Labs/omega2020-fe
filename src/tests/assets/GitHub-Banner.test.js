import React from 'react';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Banner from '../../components/assets/GitHub-Banner';
import Adapter from 'enzyme-adapter-react-16';
import { Typography, IconButton, Box } from '@material-ui/core';
import GitHubIcon from '@material-ui/icons/GitHub';

configure({ adapter: new Adapter() });

describe('the Github banner ', () => {
  const ReactWrapper = shallow(<Banner />);
  
  const renderGitBanner = renderer.create(<Banner />); 
  const rootGitBanner = renderGitBanner.root;

  it('snapshot the components', () => {
    const banner = renderer.create(<Banner />).toJSON();
    expect(banner).toMatchSnapshot();
  });

  it('renders the material ui length correctly, and the text content', () => {
    expect(ReactWrapper.find(Typography)).toHaveLength(1);
    expect(ReactWrapper.find(Typography).text()).toEqual(' Find us on GitHub');
    expect(ReactWrapper.find(IconButton)).toHaveLength(1);
    expect(ReactWrapper.find(GitHubIcon)).toHaveLength(1);
    expect(ReactWrapper.find(Box)).toHaveLength(3);
  });

  it('shows the props in the GitHubIcon', () => {
    expect(rootGitBanner.findByType(GitHubIcon).props.style).toMatchObject({"fontSize": 40, "padding": 20});
  })
});
