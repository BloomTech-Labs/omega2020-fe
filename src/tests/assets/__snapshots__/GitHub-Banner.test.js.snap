// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Github banner  snapshot the components 1`] = `
<div
  className="MuiBox-root MuiBox-root-7 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-8 makeStyles-banner-3"
  >
    <div
      className="MuiBox-root MuiBox-root-9 makeStyles-icon-2"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
        href="https://github.com/Lambda-School-Labs/omega2020-fe"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "fontSize": 40,
                "padding": 20,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </a>
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
         Find us on GitHub
      </h6>
    </div>
  </div>
</div>
`;
